Mathlib/FieldTheory/KummerExtension.lean

```diff
@@ -0,0 +1,444 @@
+noncomputable
+def autAdjoinRootXPowSubCEquiv (hn : 0 < n) :
+    rootsOfUnity ⟨n, hn⟩ K ≃* (K[n√a] ≃ₐ[K] K[n√a]) where
+  __ := autAdjoinRootXPowSubC n a hn
+  invFun := AdjoinRootXPowSubCEquivToRootsOfUnity hζ H hn
+  left_inv := by
+    intro η
+    have := Fact.mk H
+    have : IsDomain K[n√a] := inferInstance
+    letI : Algebra K K[n√a] := inferInstance
+    apply (rootsOfUnityEquivOfPrimitiveRoots
+      (n := ⟨n, hn⟩) (algebraMap K K[n√a]).injective hζ).injective
+    ext
+    simp only [algebraMap_eq, OneHom.toFun_eq_coe, MonoidHom.toOneHom_coe,
+      autAdjoinRootXPowSubC_root, Algebra.smul_def, ne_eq, MulEquiv.apply_symm_apply,
+      rootsOfUnity.val_mkOfPowEq_coe, rootsOfUnityEquivOfPrimitiveRoots_apply,
+      AdjoinRootXPowSubCEquivToRootsOfUnity]
+    split_ifs with h
+    · obtain rfl := not_imp_not.mp (fun hn ↦ ne_zero_of_irreducible_X_pow_sub_C' hn H) h
+      have : (η : Kˣ) = 1 := (pow_one _).symm.trans η.prop
+      simp only [PNat.mk_one, this, Units.val_one, map_one]
+    · exact mul_div_cancel _ (root_X_pow_sub_C_ne_zero' hn h)
+  right_inv := by
+    intro e
+    have := Fact.mk H
+    letI : Algebra K K[n√a] := inferInstance
+    apply AlgEquiv.coe_algHom_injective
+    apply AdjoinRoot.algHom_ext
+    simp only [algebraMap_eq, OneHom.toFun_eq_coe, MonoidHom.toOneHom_coe, AlgHom.coe_coe,
+      autAdjoinRootXPowSubC_root, Algebra.smul_def, PNat.mk_coe,
+      AdjoinRootXPowSubCEquivToRootsOfUnity]
+    rw [rootsOfUnityEquivOfPrimitiveRoots_symm_apply, rootsOfUnity.val_mkOfPowEq_coe]
+    split_ifs with h
+    · obtain rfl := not_imp_not.mp (fun hn ↦ ne_zero_of_irreducible_X_pow_sub_C' hn H) h
+      rw [(pow_one _).symm.trans (root_X_pow_sub_C_pow 1 a), one_mul,
+        ← algebraMap_eq, AlgEquiv.commutes]
+    · refine div_mul_cancel _ (root_X_pow_sub_C_ne_zero' hn h)
+
+lemma autAdjoinRootXPowSubCEquiv_apply (hn : 0 < n) (η) :
+    autAdjoinRootXPowSubCEquiv hζ H hn η = autAdjoinRootXPowSubC n a hn η := rfl
+
+lemma autAdjoinRootXPowSubCEquiv_symm_smul (hn : 0 < n) (σ) :
+    ((autAdjoinRootXPowSubCEquiv hζ H hn).symm σ : Kˣ) • (root _ : K[n√a]) = σ (root _) := by
+  have := Fact.mk H
+  simp only [autAdjoinRootXPowSubCEquiv, OneHom.toFun_eq_coe, MonoidHom.toOneHom_coe,
+    MulEquiv.symm_mk, MulEquiv.coe_mk, Equiv.coe_fn_symm_mk, AdjoinRootXPowSubCEquivToRootsOfUnity,
+    algebraMap_eq, Units.smul_def, Algebra.smul_def, rootsOfUnityEquivOfPrimitiveRoots_symm_apply,
+    rootsOfUnity.mkOfPowEq, PNat.mk_coe, Units.val_ofPowEqOne, ite_mul, one_mul, ne_eq]
+  simp_rw [← root_X_pow_sub_C_eq_zero_iff H]
+  split_ifs with h
+  · rw [h, mul_zero, map_zero]
+  · rw [div_mul_cancel _ h]
+
+end AdjoinRoot
+
+/-! ### Galois Group of `IsSplittingField K L (X ^ n - C a)` -/
+
+section IsSplittingField
+
+variable {n} {a}
+variable {L : Type*} [Field L] [Algebra K L] [IsSplittingField K L (X ^ n - C a)]
+
+lemma isSplittingField_AdjoinRoot_X_pow_sub_C :
+    haveI := Fact.mk H
+    letI : Algebra K K[n√a] := inferInstance
```

Comment:
It would be interesting to understand why this line is needed...

Reply:
It times out otherwise. I think it's one of the problems smart lean4 people like Anne Baanen are working on.